nilearn/decoding/fista.py

```diff
@@ -184,7 +187,9 @@ def mfista(f1_grad, f2_prox, total_energy, lipschitz_constant, w_size,
 
         # backward (prox) step
         for _ in range(10):
-            w, prox_info = f2_prox(z - stepsize * gradient_buffer, stepsize,
+            # z <- -stepsize * gradient_buffer + z
+            z = inplace_add_mult(gradient_buffer, z, a=-stepsize)
```

Comment:
makes sense - but we have full control over z as a local variable. Why
would we let that happen? Or alternatively, shouldn't we warn about
performance loss if this situation occurs?

On Friday, September 4, 2015, Gael Varoquaux <notifications@github.com>
wrote:

> In nilearn/decoding/fista.py
> <https://github.com/nilearn/nilearn/pull/774#discussion_r38721405>:
>
> > @@ -184,7 +187,9 @@ def mfista(f1_grad, f2_prox, total_energy, lipschitz_constant, w_size,
> >
> >          # backward (prox) step
> >          for _ in range(10):
> > -            w, prox_info = f2_prox(z - stepsize * gradient_buffer, stepsize,
> > +            # z <- -stepsize * gradient_buffer + z
> > +            z = inplace_add_mult(gradient_buffer, z, a=-stepsize)
>
> I recall from the ADMM version that this assignment to z was necessary
> there, too. If this operation really is in place, it wouldn't need the
> assignment to z. Does it work without it?
> Actually, if z is not c-contiguous, there can be an implicit copy.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/774/files#r38721405>.
>